proto-public/annotations/ratelimit/ratelimit.proto

```diff
@@ -0,0 +1,22 @@
+syntax = "proto3";
+
+import "google/protobuf/descriptor.proto";
+
+package hashicorp.consul.internal.ratelimit;
+
+// OperationType determines the kind of rate limit that will be applied to this
+// RPC (i.e. read or write).
+enum OperationType {
+  UNSPECIFIED = 0;
+  READ = 1;
+  WRITE = 2;
+}
+
+// Spec describes the kind of rate limit that will be applied to this RPC.
+message Spec {
+  OperationType operation_type = 1;
+}
+
+extend google.protobuf.MethodOptions {
+  Spec spec = 8300;
```

Comment:
This number is arbitrary, it just needs to not conflict with other extensions. I chose it because it's our server port 😄